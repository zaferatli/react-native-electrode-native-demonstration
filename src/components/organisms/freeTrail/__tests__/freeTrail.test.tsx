import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';

import FreeTrail, {FreeTrailPropsType} from '../index';

const getComponent = (props: FreeTrailPropsType) => {
  return <FreeTrail {...props} />;
};

it('initial render button', () => {
  renderer.create(
    getComponent({
      value: true,
    }),
  );
});

it('Primary FreeTrail', () => {
  const props: FreeTrailPropsType = {
    value: false,
  };
  const RenderedComponent = renderer.create(getComponent(props)).toJSON();
  expect(RenderedComponent).toMatchSnapshot();
});
