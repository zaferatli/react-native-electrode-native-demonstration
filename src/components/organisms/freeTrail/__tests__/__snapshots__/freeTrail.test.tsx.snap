// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary FreeTrail 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#FFF5F5FF",
        "borderRadius": 8,
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": undefined,
        "height": 60,
        "justifyContent": "space-between",
        "marginTop": 15,
        "paddingHorizontal": 20,
        "width": 361,
      },
      Object {
        "borderColor": "#E7E8E9FF",
        "borderRadius": 10,
        "borderWidth": 1,
      },
      undefined,
    ]
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#717578FF",
          "fontSize": 12,
          "fontStyle": "normal",
          "fontWeight": "500",
          "lineHeight": undefined,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#000000FF",
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "500",
          "lineHeight": undefined,
        }
      }
    />
  </View>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "rgba(255, 0, 0, 1)",
        "borderRadius": 14,
        "height": 31,
        "position": "relative",
        "width": 51,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderRadius": 14,
            "height": 27,
            "left": 2,
            "position": "absolute",
            "top": 2,
            "width": 27,
          }
        }
      />
    </View>
  </View>
</View>
`;
